src/library/LibraryWindow.vala

```diff
@@ -1185,10 +1184,6 @@ public class LibraryWindow : AppWindow {
         }
 
         set_metadata_sidebar_visible (is_metadata_sidebar_visible ());
-
-        if (get_current_page () != null) {
-            get_current_page ().grab_focus ();
-        }
     }
```

Comment:
This may have accessibility implications - under most circumstances the welcome page should be focussed.
I suggest instead adding another parameter `bool grab_focus = true` and then
```
        if (get_current_page () != null && grab_focus) {
            get_current_page ().grab_focus ();
        }
```

and then  altering the call to this function from `update_view_filter_message ()`  accordinglingy

```
window.toggle_welcome_page (false, "", "", false, false);
```
This will avoid unrelating consequences

Reply:
I think that is overcomplicating things. We don't try to move focus around like this in any other app. If the user has explicitly chosen to focus some widget, we shouldn't be trying to move around focus under their feet.

Reply:
By pressing Enter in the search the user is intending to end the search so I think the expected behaviour is to refocus the view.

I'll have to investigate the circumstances of showing the welcome page further but that wasn' the subject of the issue.